app/models/qernel/plugins/merit/producer_adapter.rb

```diff
@@ -42,23 +35,29 @@ def inject!
       def producer_attributes
         attrs = super
 
-        attrs[:marginal_costs] = @converter.marginal_costs
-
-        attrs[:output_capacity_per_unit] =
-          @converter.electricity_output_conversion * @converter.input_capacity
-
-        attrs[:fixed_costs_per_unit] =
-          @converter.send(:fixed_costs)
-
-        attrs[:fixed_om_costs_per_unit] =
-          @converter.send(:fixed_operation_and_maintenance_costs_per_year)
+        attrs[:marginal_costs] = marginal_costs
+        attrs[:output_capacity_per_unit] = output_capacity_per_unit
 
         attrs
       end
 
       def producer_class
         ::Merit::DispatchableProducer
       end
+
+      def marginal_costs
+        @converter.marginal_costs
+      end
+
+      def output_capacity_per_unit
+        @converter.electricity_output_conversion * @converter.input_capacity
+      end
+
+      # Internal: Capacity used to multiply full load seconds to determien the
+      # resulting annual demand of the producer.
+      def flh_capacity
```

Comment:
For producers, the input capacity is used for the flh_capacity. Why is this different from how the import interconnector is treated?

Reply:
This is used to convert the FLH of the Merit producer back to a demand for ETEngine. It's not a very good explanation, but input_capacity is used because [that's what ETE has always used](https://github.com/quintel/etengine/commit/969da27ca80785c930199d4f85e03c49ba1fc115#diff-bd4da10f28ee1072afe37359a7f64ff7R62) (this is not a new addition; it's just moving what was already there to a new method so that the class is more flexible).